### ISSUE TITLE
Deprecated DynamicLinks and Auth

### ISSUE THREAD

**Issue by paulb777** [firebase/firebase-ios-sdk#12738]
Now Fixed:

**See migration guide section "Configure your project to use the new links." https://firebase.google.com/docs/auth/ios/email-link-migration.**

Opened this issue for tracking Firebase Auth transitioning from its deprecated DynamicLinks dependency

### Discussed in https://github.com/firebase/firebase-ios-sdk/discussions/12726

<div type='discussions-op-text'>

<sup>Originally posted by **sergiocampama** April  8, 2024</sup>
It is our understanding that Auth sends email verification links and password resets using DynamicLinks. What will happen to those when DynamicLinks is removed later this year? Is there a plan for that?</div>

**Comment by paulb777** [comment]
From https://github.com/firebase/firebase-ios-sdk/issues/12556#issuecomment-2005211499:

Our team is aware of Firebase Dynamic Link deprecation. We're working to ensure sendEmailVerification and other Auth methods continue to function after FDL's shutdown. Meanwhile, you can enable FDL in the Firebase Console to address the DYNAMIC_LINK_NOT_ACTIVATED error.

**Comment by ueman** [comment]
The button to enable Firebase Dynamic Links has been replaced by a deprecation notice. How am I supposed to enable Dynamic Links?

**Comment by NhienLam** [comment]
Hi @ueman, could you please [open a Firebase support ticket](https://firebase.google.com/support/troubleshooter/contact)? They would be in a better place to help you. Thanks!

**Comment by s-chandel** [comment]
Hi @ueman, if you enable email link authentication from the Firebase Authentication section, the buttons for configuring Firebase Dynamic Links should appear in the FDL section.

**Comment by ErionTp** [comment]
<img width="386" alt="Screenshot 2024-07-19 at 18 54 17" src="https://github.com/user-attachments/assets/7ffbbb9b-4b15-4dc9-a46c-aa11ef545a7a">

<img width="764" alt="Screenshot 2024-07-19 at 18 54 58" src="https://github.com/user-attachments/assets/05023c69-563c-4f28-8803-d17ea1a54c37">

I cannot activate dynamic links, any suggestions please?

**Comment by alexandrubichis** [comment]
Is there any update on this issue, or an estimate of when it will be resolved?

**Comment by tensoralex** [comment]
+1, the issue still exists for IOS sdk, all others (web, android) do not require dynamic links anymore for email auth.
Only the IOS SDK is not working without dynamic links. 
I have removed dynamic links in the Firebase console, and now there is no way to put it back. 
Email auth on IOS is broken.

**Comment by olerhan** [comment]
Hi, I’m facing the same issue and after extensive research, I came across this issue. Is there any update?

**Comment by ViktorSimko** [comment]
Hi, 

Could you please share an update on this? I planned to use Firebase Auth for my project but if this won't get resolved the only option I'll have is to choose another provider.

**Comment by AbhishekDoshi26** [comment]
Hello team!

Any updates on the same? Still waiting for a resolution

**Comment by Xiaoshouzi-gh** [comment]
Firebase Auth team is actively working on this.  Please stay tuned.

**Comment by nerder** [comment]
Hey @Xiaoshouzi-gh , any expected release date for this one? We have couple of user stories pending this to be played

**Comment by nerder** [comment]
Circling back on this, is there any news?

**Comment by paulb777** [comment]
Yes. 11.8.0 released this week with a deprecation of `dynamicLinkDomain` in favor of the added `ActionCodeSettings.linkDomain` to customize the Firebase Hosting link domain that is used in out-of-band email action flows.

**Comment by joetam** [comment]
Hi @paulb777, i'm on 11.8.1 and this does not work. Without a single link dynamic link configured on my project, and not setting `dynamicLinkDomain` results in this error being thrown in the SDK:

```
{
    FIRAuthErrorUserInfoNameKey = "ERROR_INTERNAL_ERROR";
    NSLocalizedDescription = "An internal error has occurred, print and inspect the error details for more information.";
    NSUnderlyingError = "Error Domain=FIRAuthInternalErrorDomain Code=3 \"(null)\" UserInfo={NSUnderlyingError=0x600000cf94a0 {Error Domain=com.google.HTTPStatus Code=400 \"(null)\" UserInfo={data={length = 288, bytes = 0x7b0a2020 22657272 6f72223a 207b0a20 ... 5d0a2020 7d0a7d0a }, data_content_type=application/json; charset=UTF-8}}, FIRAuthErrorUserInfoDeserializedResponseKey={\n    code = 400;\n    errors =     (\n                {\n            domain = global;\n            message = \"DYNAMIC_LINK_NOT_ACTIVATED : FDL domain is not configured\";\n            reason = invalid;\n        }\n    );\n    message = \"DYNAMIC_LINK_NOT_ACTIVATED : FDL domain is not configured\";\n}}";
}
```

**Comment by ViktorSimko** [comment]
Hi @paulb777 ,

I'm also getting the same error on 11.8.1 as @joetam above.

**Comment by tensoralex** [comment]
Same for us. Doesn't work.

**Comment by paulb777** [comment]
@Xiaoshouzi-gh Would you take a look?

**Comment by joetam** [comment]
It started working for me once I have made the corresponding project config using the `firebase-admin` SDK. See the section "Configure your project to use the new links." https://firebase.google.com/docs/auth/ios/email-link-migration.

It would be nice if this was handled for us, especially since the client code is not using any dynamic links.

**Comment by tensoralex** [comment]
> It started working for me once I have made the corresponding project config using the `firebase-admin` SDK. See the section "Configure your project to use the new links." https://firebase.google.com/docs/auth/ios/email-link-migration.
> 
> It would be nice if this was handled for us, especially since the client code specified to use the new links.

```
const updateRequest = {
mobileLinksConfig: {
    domain: 'HOSTING_DOMAIN'
}
}
const updateProjectConfig = () => {
projectConfigManager.updateProjectConfig(updateRequest)
.then((response) => {
    // updated project config
}).catch((error) => {
    console.log('Error updating the project:', error);
});
}

```
@joetam we are getting 404s when attempting to update the configuration, any suggestion which firebase-admin endpoint you called to updated it?

**Comment by tensoralex** [comment]
> > It started working for me once I have made the corresponding project config using the `firebase-admin` SDK. See the section "Configure your project to use the new links." https://firebase.google.com/docs/auth/ios/email-link-migration.
> > It would be nice if this was handled for us, especially since the client code specified to use the new links.
> 
> ```
> const updateRequest = {
> mobileLinksConfig: {
>     domain: 'HOSTING_DOMAIN'
> }
> }
> const updateProjectConfig = () => {
> projectConfigManager.updateProjectConfig(updateRequest)
> .then((response) => {
>     // updated project config
> }).catch((error) => {
>     console.log('Error updating the project:', error);
> });
> }
> ```
> 
> [@joetam](https://github.com/joetam) we are getting 404s when attempting to update the configuration, any suggestion which firebase-admin endpoint you called to updated it?

Nevermind. This worked:
```
const projectConfigManager = admin.auth().projectConfigManager();

const updateRequest = {
    mobileLinksConfig: {
        domain: 'HOSTING_DOMAIN'
    }
};

const updateProjectConfig = () => {
    projectConfigManager.updateProjectConfig(updateRequest)
        .then((response) => {
            console.log("Project configuration updated successfully:", response);
        })
        .catch((error) => {
            console.log('Error updating the project:', error);
        });
}

updateProjectConfig();
```

**Comment by AbhishekDoshi26** [comment]
Do we have JAVA equivalent for the same?

**Comment by JS-GitRepo** [comment]
> It started working for me once I have made the corresponding project config using the `firebase-admin` SDK. See the section "Configure your project to use the new links." https://firebase.google.com/docs/auth/ios/email-link-migration.
> 
> It would be nice if this was handled for us, especially since the client code specified to use the new links.

I'm not too familiar with how Hosting fits into the equation here; It sounds like you set up a hosting project and run this code snippet on it:

```js
const updateRequest = {
mobileLinksConfig: {
    domain: 'HOSTING_DOMAIN'
}
}
const updateProjectConfig = () => {
projectConfigManager.updateProjectConfig(updateRequest)
.then((response) => {
    // updated project config
}).catch((error) => {
    console.log('Error updating the project:', error);
});
}
```

But where does this code actually live? What file do I put it in? I don't have any Hosting files made outside of what was deployed directly from the firebase CLI.

**Comment by tensoralex** [comment]
> > It started working for me once I have made the corresponding project config using the `firebase-admin` SDK. See the section "Configure your project to use the new links." https://firebase.google.com/docs/auth/ios/email-link-migration.
> > It would be nice if this was handled for us, especially since the client code specified to use the new links.
> 
> I'm not too familiar with how Hosting fits into the equation here; It sounds like you set up a hosting project and run this code snippet on it:
> 
> const updateRequest = {
> mobileLinksConfig: {
>     domain: 'HOSTING_DOMAIN'
> }
> }
> const updateProjectConfig = () => {
> projectConfigManager.updateProjectConfig(updateRequest)
> .then((response) => {
>     // updated project config
> }).catch((error) => {
>     console.log('Error updating the project:', error);
> });
> }
> 
> But where does this code actually live? What file do I put it in? I don't have any Hosting files made outside of what was deployed directly from the firebase CLI.

No, this is a one-time update for the Firebase settings through Firebase admin API (they don't have a UI setting yet for this, per my understanding). 
You can run it on your local machine if you install Firebase admin SDK: https://www.npmjs.com/package/firebase-admin

**Comment by tensoralex** [comment]
> But where does this code actually live? What file do I put it in? I don't have any Hosting files made outside of what was deployed directly from the firebase CLI.

You can run this on your local machine:

```
// Make sure to install firebase-admin with: npm install firebase-admin
const admin = require('firebase-admin');

// Adjust the path below for your firebase-admin-key.json 
const serviceAccount = require("../path/firebase-admin-key.json");

// Initialize the Firebase Admin SDK with the service account credentials.
admin.initializeApp({
    credential: admin.credential.cert(serviceAccount)
});

const projectConfigManager = admin.auth().projectConfigManager();

const updateRequest = {
    mobileLinksConfig: {
        domain: 'HOSTING_DOMAIN'
    }
};

const updateProjectConfig = () => {
    projectConfigManager.updateProjectConfig(updateRequest)
        .then((response) => {
            console.log("Project configuration updated successfully:", response);
        })
        .catch((error) => {
            console.log('Error updating the project:', error);
        });
}

updateProjectConfig();
```
